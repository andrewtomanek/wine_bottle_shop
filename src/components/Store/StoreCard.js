import React from "react";
import classes from "./StoreCard.module.css";

const StoreCard = ({ itemName, itemType }) => {
  return (
    <div className={`${classes[itemType]} ${classes.ShopItem}`}>
      <p className={classes.ItemLabel}>{itemName}</p>
    </div>
  );
};

export default StoreCard;
