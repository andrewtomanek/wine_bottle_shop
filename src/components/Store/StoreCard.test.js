import React from "react";
import { shallow } from "enzyme";
import StoreCard from "./StoreCard";

const props = {
  itemType: "shopItem1",
  itemName: "TestName",
};

describe("Shallow Render StoreCard component", () => {
  it("renders item name", () => {
    const wrapper = shallow(<StoreCard {...props} />);
    expect(wrapper.find("p").text()).toContain("TestName");
  });

  it("Adds correct shop item class", () => {
    const wrapper = shallow(<StoreCard {...props} />);
    expect(wrapper.find({ className: "shopItem1 ShopItem" })).toHaveLength(1);
  });

  it("Match Snapshot of StoreCard component", () => {
    const wrapper = shallow(<StoreCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
