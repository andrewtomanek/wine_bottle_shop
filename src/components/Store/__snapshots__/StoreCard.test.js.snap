// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Render StoreCard component Match Snapshot of StoreCard component 1`] = `
<div
  className="shopItem1 ShopItem"
>
  <p
    className="ItemLabel"
  >
    TestName
  </p>
</div>
`;
